docs/odh/oci-model-storage.md

```diff
@@ -8,15 +8,227 @@ which also explains how to deploy models from OCI images.
 
 This page offers a guide similar to the upstream project documentation, but
 focusing on the OpenDataHub and OpenShift characteristics. To demonstrate
-how to create an OCI container image, the publicly available [MobileNet v2-7
-model](https://github.com/onnx/models/tree/main/validated/vision/classification/mobilenet)
-is used. This model is in ONNX format.
+how to create and use OCI containers, two examples are provided:
+* The first example uses [IBM's Granite-3.0-2B-Instruct model](https://huggingface.co/ibm-granite/granite-3.0-2b-instruct) 
+  available in Hugging Face. This is a generative AI model.
+* The second example uses the [MobileNet v2-7 model](https://github.com/onnx/models/tree/main/validated/vision/classification/mobilenet)
+  is used. This is a predictive AI model in ONNX format.
+
+## Creating and deploying an OCI image of IBM's Granite-3.0-2B-Instruct model
+
+IBM's Granite-3.0-2B-Instruct model is [available at Hugging Face](https://huggingface.co/ibm-granite/granite-3.0-2b-instruct).
+To create an OCI container image, the model needs to be downloaded and copied into
+the container. Once the OCI image is built and published in a registry, it can be
+deployed on the cluster.
+
+The ODH projects provides configurations for the vLLM model server, which
+supports running the Granite model. Thus, this guide will use this model server
+to demonstrate how to deploy the Granite model stored in an OCI image.
+
+### Storing the Granite model in an OCI image
+
+To download the Granite model, the [`huggingface-cli download` command](https://huggingface.co/docs/huggingface_hub/guides/cli#huggingface-cli-download) can be used.
+The process is as follows:
+* Install the huggingface CLI
+* Use huggingface CLI to download the model
+* Create the final OCI using the downloaded model
+
+It is possible to run this process as part of a multi-stage container build. Create a file named
+`Containerfile` with the following contents:
+```Dockerfile
+##### Stage 1: Download the model
+FROM registry.access.redhat.com/ubi9/python-312:latest as downloader
+
+# Install huggingface-cli
+RUN pip install "huggingface_hub[cli]"
+
+# Download the model
+ARG repo_id
+RUN mkdir models && huggingface-cli download --quiet --max-workers 2 --local-dir ./models $repo_id
+
+##### Stage 2: Build the final OCI model container
+FROM registry.access.redhat.com/ubi8/ubi-micro:latest as model
+
+# Copy from the download stage
+COPY --from=downloader --chown=0:0 /opt/app-root/src/models /models
```

Comment:
On OpenShift, convention is to use default user 1001 and group 0. Setting user 0 does not have real impact as group 0 will override it. However this may be an issue if people want to use this container in another context, like locally with Podman.
I would switch to 1001:0

Reply:
I've changed all instances to user 1001.